src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -452,6 +453,9 @@ internal async Task HandleSyntaxParseRequest(SyntaxParseParams param, RequestCon
                 }
                 else
                 {
+                    if (RefreshNeeded(scriptFile.ClientUri)) {
```

Comment:
As we chatted before, I think we should let ConnectionService handle the refresh automatically. for example when the connection is added, if it has expiresOn info, then we will automatically refresh it when it is about to expire. when the connection is removed we will delete the auto refresh handler.

Reply:
We chatted about this in our team sync today, and it will have to be a two part plan - first we will get this PR in to get the customer unblocked, and then I will circle back and implement a more global solution for the token refresh.

Reply:
I am fine with it, please create an issue to track it.

Reply:
this code block should be defined as a method and it seems you are not using the output of the method, what if the refresh fails?